### ISSUE TITLE
Add ability to specify import path instead of hardcode to `tonic-build` codegen

### ISSUE THREAD

**Issue by evergreen-trading-systems** [hyperium/tonic#699]
## Feature Request
Add ability to specify path to `tonic` and `async_trait` when referenced in codegen.
### Crates
tonic-build

### Motivation
I keep `tonic` and many other dependencies in a separate package. When I import them into a package that uses `tonic-build` to generate Rust proto, I get an error saying `tonic` and `async_trait` cannot be found.

Currently, the only way to use `tonic` is a direct dependency of the main package not as an import from another package. Haven't come across this issue with other packages. Serde has "crate" container attribute that solves it.

### Proposal
Add a new method to the `tonic_build::configure() -> Config Builder` trait.

### Alternatives

The alternative way is I clone the repo and hardcode the correct import path.

**Comment by davidpdrsn** [comment]
I don't follow. Why can't you depend on tonic for the crate that includes the compiled protos?

**Comment by evergreen-trading-systems** [comment]
If I import `tonic` and `prost` from another package, the codegen does not find the modules
![image](https://user-images.githubusercontent.com/4870868/124825198-38a89880-df6b-11eb-934e-bacfe04ec485.png)

However, if I include `tonic` and `prost` in the same package that compiles protos, it works
![image](https://user-images.githubusercontent.com/4870868/124825994-3abf2700-df6c-11eb-8be0-b050f7a30bcf.png)

I don't know why that is.

**Comment by davidpdrsn** [comment]
I suppose that is because the client and server codegen nests things in modules. See https://github.com/hyperium/tonic/blob/master/tonic-build/src/server.rs#L67 and https://github.com/hyperium/tonic/blob/master/tonic-build/src/client.rs#L25.

I wonder if adding `use super::*` to the generated modules would fix it or if a `crate` config similar to serde's is necessary.

**Comment by evergreen-trading-systems** [comment]
Yes if I refer to tonic as `crate::tonic::codegen::*`, it works but `prost` doesn't. 
It uses `::` naked which isn't going to work in this Rust-2018 project according to https://doc.rust-lang.org/reference/visibility-and-privacy.html#pubin-path-pubcrate-pubsuper-and-pubself

**Comment by LucioFranco** [comment]
Do you have a simple reproduction of your issue? I am not totally following what the issue is.

**Comment by JSAbrahams** [comment]
Just chiming in, a use case for this I think is also avoiding have to align tonic versions across many crates.
Say I have the following:

- `util` crate, with `tonic` version `x`, and which re-exports `tonic`, and with some wrapper functions around tonic.
- crate `A`, which has `tonic` version `x`, and which depends on `util`
- crate `B`, which has `tonic` version `x`, and which depends on `util`

Now, if I wish to update the `tonic` version in `util`, I have to make sure that crate 'A` and crate `B`'s versions are also aligned.
A much more elegant solution would be if I could just depend on the re-export `tonic` from `util` as such `use util::tonic` (in say crate `A`). 
But since `tonic_build` outputs files where `tonic::` is hardcoded, the compiler does not allow this.

In this example, ideally, I would like to be able to tell `tonic_build` to output files where we instead of `util::tonic`, for instance.

**Comment by lukasweber** [comment]
Hi, just created a PR to fix this issue by proposing to use a similar strategy as `prost` is using with [prost_path](https://docs.rs/prost-build/latest/prost_build/struct.Config.html#method.prost_path). I suggest to add a `tonic_path` method to the code gen builder which allows to configure the path to the `tonic` crate. Let me know what you think about it.

If you look at the `tests/` folder in my changeset, you can see how it would look like.
